mimic/core.py

```diff
@@ -51,9 +52,25 @@ def __init__(self, clock, apis, domains=()):
         self.domains = list(domains)
 
         for api in apis:
-            this_api_id = ((api.__class__.__name__) + '-' +
-                           random_hex_generator(3))
-            self._uuid_to_api[this_api_id] = api
+            # Gate check the API to make sure it implements one of the
+            # supported interfaces
+            required_interfaces = (IAPIMock, IExternalAPIMock)
+            provided_interfaces = list(zope.interface.providedBy(api))
+            if True in [i in provided_interfaces
```

Comment:
I'll actually be refactoring this a little in another PR. I think the final proposal makes sense to do.

The two things that will happen later is the entire body of the loop will become an `add` function call, and then the interface check might also become a `validate` function used by the `add` function (not sure yet on the validate function part).